packages/wallet-sdk/src/components/RedirectDialog/RedirectDialog.tsx

```diff
@@ -0,0 +1,67 @@
+import clsx from "clsx";
+import { FunctionComponent, h, render } from "preact";
+
+import { injectCssReset } from "../../lib/cssReset";
+import { SnackbarContainer } from "../Snackbar";
+import css from "./RedirectDialog-css";
+
+type RedirectDialogProps = {
+  title: string;
+  buttonText: string;
+  onButtonClick: () => void;
+};
+
+export class RedirectDialog {
+  private root: Element | null = null;
+
+  public attach(): void {
+    const el = document.documentElement;
+    this.root = document.createElement("div");
+    this.root.className = "-cbwsdk-css-reset";
+    el.appendChild(this.root);
+    injectCssReset();
+  }
+
+  public present(props: RedirectDialogProps): void {
+    this.render(props);
+  }
+
+  public clear(): void {
+    this.render(null);
+  }
+
+  private render(props: RedirectDialogProps | null): void {
+    if (!this.root) return;
+    render(null, this.root);
+
+    if (!props) return;
+    render(
+      <RedirectDialogContent
+        {...props}
+        onDismiss={() => {
+          this.clear();
+        }}
+      />,
+      this.root,
+    );
+  }
+}
```

Comment:
This is confusing. we initially render null then check for props then return again

could we just do this

```suggestion
  private render(props: RedirectDialogProps | null): void {
    if (!this.root || !props) return;
    render(
      <RedirectDialogContent
        {...props}
        onDismiss={() => {
          this.clear();
        }}
      />,
      this.root,
    );
  }
}

```

Reply:
i found it needs to call `render(null, this.root);` here, otherwise sometimes it would stack multiple instances of this dialog.
note that, without proper refactor of WalletLinkRelay, it doesn't seem straightfoward to fine tune its `this.ui` behavior.

Reply:
interesting. Okay to leave it for now but something we should address in future versions